test/spec_server.rb

```diff
@@ -334,6 +334,28 @@ def start(*) [self.class, :started] end
     c.start.must_equal [c, :started]
   end
 
+  def start_server(server)
+    begin
+      ready = Thread::Queue.new
+      thread = Thread.new do
+        begin
+          server.start { |s| ready.push(true) }
+        ensure
+          ready.close
+        end
+      end
+
+      unless ready.pop
+        thread.join
+      end
+
+      sleep 10
```

Comment:
This seems hacky, sleep forever or not sleep seems more reliable

Reply:
It's a total hack to see if it's a race condition or something else.